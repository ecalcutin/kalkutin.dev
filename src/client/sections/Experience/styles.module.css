.container {
  --tab-width: 15rem;
  --tab-height: 4rem;

  display: flex;

  @media (max-width: 768px) {
    display: block;
  }
}

.section {
  max-width: 70rem;
}
