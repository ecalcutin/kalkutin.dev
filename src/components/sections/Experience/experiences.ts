export default [
  {
    company: "Endava",
    position: "Engineer",
    dateRange: "JUN 2018 - June 2020",
    url: "https://www.endava.com/",
  },
  // {
  //   company: "Endava",
  //   position: "Engineer",
  //   dateRange: "JUN 2018 - June 2020",
  //   url: "https://www.endava.com/",
  // },
];
