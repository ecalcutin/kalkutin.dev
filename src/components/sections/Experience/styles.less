@import (reference) "theme/mixins.less";

.experience {
  max-width: 700px;

  .inner {
    display: flex;

    @media (max-width: 600px) {
      display: block;
    }

    // Prevent container from jumping
    @media (min-width: 700px) {
      min-height: 340px;
    }
  }
}

.tabList {
  position: relative;
  z-index: 3;
  width: max-content;
  padding: 0;
  margin: 0;
  list-style: none;

  @media (max-width: 600px) {
    display: flex;
    overflow-x: auto;
    width: calc(100% + 100px);
    padding-left: 50px;
    margin-left: -50px;
    margin-bottom: 30px;
  }

  @media (max-width: 480px) {
    width: calc(100% + 50px);
    padding-left: 25px;
    margin-left: -25px;
  }

  li {
    &:first-of-type {
      @media (max-width: 600px) {
        margin-left: 50px;
      }
      @media (max-width: 480px) {
        margin-left: 25px;
      }
    }
    &:last-of-type {
      @media (max-width: 600px) {
        padding-right: 50px;
      }
      @media (max-width: 480px) {
        padding-right: 25px;
      }
    }
  }
}

.tabButton {
  display: flex;
  align-items: center;
  width: 100%;
  height: 42px;
  padding: 0 20px 2px;
  border-left: 2px solid @color-primary-lighten;
  background-color: transparent;
  color: @color-secondary;
  display: inline-block;
  text-decoration: none;
  text-decoration-skip-ink: auto;
  color: inherit;
  position: relative;
  transition: @transition-all;
  &:hover,
  &:active,
  &:focus {
    color: @color-accent;
    outline: 0;
  }
  //   color: ${({ isActive }) => (isActive ? 'var(--green)' : 'var(--slate)')};
  font-family: @font-mono;
  font-size: @font-size-xs;
  text-align: left;
  white-space: nowrap;

  @media (max-width: 768px) {
    padding: 0 15px 2px;
  }

  @media (max-width: 600px) {
    .flexCenter();
    min-width: 120px;
    padding: 0 15px;
    border-left: 0;
    border-bottom: 2px solid @color-primary-lighten;
    text-align: center;
  }

  &:hover,
  &:focus {
    background-color: @color-primary-light;
  }
}

.highlight {
  position: absolute;
  top: 0;
  left: 0;
  z-index: 10;
  width: 2px;
  height: 42px;
  border-radius: @border-radius;
  background: @color-accent;
  //   transform: translateY(
  //     calc($ {({activeTabId}) => activeTabId} * var(--tab-height))
  //   );
  transition: transform 0.25s cubic-bezier(0.645, 0.045, 0.355, 1);
  transition-delay: 0.1s;

  @media (max-width: 600px) {
    top: auto;
    bottom: 0;
    width: 100%;
    max-width: 120px;
    height: 2px;
    margin-left: 50px;
    // transform: translateX(
    //   calc($ {({activeTabId}) => activeTabId} * var(--tab-width))
    // );
  }
  @media (max-width: 480px) {
    margin-left: 25px;
  }
}

.tabPanels {
  position: relative;
  width: 100%;
  margin-left: 20px;

  @media (max-width: 600px) {
    margin-left: 0;
  }
}

.tabPanel {
  width: 100%;
  height: auto;
  padding: 10px 5px;

  ul {
    // ${({ theme }) => theme.mixins.fancyList};
  }

  h3 {
    margin-bottom: 2px;
    font-size: @font-size-xxl;
    font-weight: 500;
    line-height: 1.3;

    .company {
      color: @color-accent;
    }
  }

  .range {
    margin-bottom: 25px;
    color: @color-secondary-light;
    font-family: @font-mono;
    font-size: @font-size-xs;
  }
}
