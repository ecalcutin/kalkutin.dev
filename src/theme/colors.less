@import "variables.less";

@color-primary-dark: darken(@color-primary, 5%);
@color-primary-light: lighten(@color-primary, 5%);
@color-primary-lighten: lighten(@color-primary, 10%);
@color-primary-shadow: darken(@color-primary, 30%);

@color-secondary-dark: darken(@color-secondary, 5%);
@color-secondary-light: lighten(@color-secondary, 5%);
@color-secondary-lighten: lighten(@color-secondary, 20%);

@color-accent-tint: fade(@color-accent, 10%);
