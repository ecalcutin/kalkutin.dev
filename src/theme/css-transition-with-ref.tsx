import React, { cloneElement, ReactElement, useRef } from 'react';
import { CSSTransition } from 'react-transition-group';
import { CSSTransitionProps } from 'react-transition-group/CSSTransition';

type Props = CSSTransitionProps & {
  readonly children: ReactElement<any>;
};

export const CSSTransitionWithRef: React.FC<Props> = props => {
  const { children, ...restProps } = props;

  const ref = useRef<HTMLElement>(null);

  const setRef = (value: HTMLElement | null): void => {
    ref.current = value;
  };

  return (
    // @ts-ignore
    <CSSTransition nodeRef={ref} {...restProps}>
      {cloneElement(children, { ref: setRef })}
    </CSSTransition>
  );
};
