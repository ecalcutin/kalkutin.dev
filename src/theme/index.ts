export { default as theme } from './theme-styles';
export { default as GlobalStyle } from './global';
export { default as mixins } from './mixins';
