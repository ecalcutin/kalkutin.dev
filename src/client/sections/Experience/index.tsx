import React, { useState } from 'react';

import { Section } from 'client/components/Section';
import { experiences } from 'client/constants/experience';

import { ExperienceTabList } from './components/ExperienceTabList';
import { ExperienceTabPanels } from './components/ExperienceTabPanels';
import styles from './styles.module.css';

export const Experience: React.FC = () => {
  const [activeTabId, setActiveTabId] = useState<number>(0);

  return (
    <Section
      id='experience'
      title='Where I have worked'
      className={styles.section}
    >
      <div className={styles.container}>
        <ExperienceTabList
          onClick={setActiveTabId}
          activeIndex={activeTabId}
          items={experiences.map(e => e.company)}
        />
        <ExperienceTabPanels
          activeTabPanel={activeTabId}
          experiences={experiences}
        />
      </div>
    </Section>
  );
};
