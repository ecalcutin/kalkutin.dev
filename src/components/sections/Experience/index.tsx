import React from "react";

import styles from "./styles.less";

const Experiece = () => {
  return (
    <section className={styles.experience}>
      <h2 className="numbered-heading">Where Iâ€™ve Worked</h2>
    </section>
  );
};

export default Experiece;
