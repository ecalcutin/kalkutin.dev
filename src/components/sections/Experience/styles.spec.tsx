import renderer from "react-test-renderer";
import { ThemeProvider } from "styled-components";

import { theme } from "../../../theme";

import Experience from "./index";

describe("Experience section", () => {
  it("should render correctly", () => {
    const tree = renderer.create(
      <ThemeProvider theme={theme}>
        <Experience />
      </ThemeProvider>
    );
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
