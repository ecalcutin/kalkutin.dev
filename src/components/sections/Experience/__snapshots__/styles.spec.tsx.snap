// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Experience section should render correctly 1`] = `
<section
  className="sc-bdfBQB jkLvXf"
  id="experience"
>
  <h2
    className="numbered-heading"
  >
    Where I’ve Worked
  </h2>
  <div
    className="inner"
  >
    <div
      aria-label="Job tabs"
      className="sc-gsTEea gYLxuY"
      role="tablist"
    >
      <button
        aria-controls="panel-0"
        aria-selected={true}
        className="sc-dlfnuX lnfZHq"
        id="tab-0"
        onClick={[Function]}
        role="tab"
        tabIndex={0}
      >
        <span>
          Grid Dynamics
        </span>
      </button>
      <button
        aria-controls="panel-1"
        aria-selected={false}
        className="sc-dlfnuX egPkXS"
        id="tab-1"
        onClick={[Function]}
        role="tab"
        tabIndex={-1}
      >
        <span>
          Qwil Messenger
        </span>
      </button>
      <button
        aria-controls="panel-2"
        aria-selected={false}
        className="sc-dlfnuX egPkXS"
        id="tab-2"
        onClick={[Function]}
        role="tab"
        tabIndex={-1}
      >
        <span>
          Code Factory
        </span>
      </button>
      <button
        aria-controls="panel-3"
        aria-selected={false}
        className="sc-dlfnuX egPkXS"
        id="tab-3"
        onClick={[Function]}
        role="tab"
        tabIndex={-1}
      >
        <span>
          Endava
        </span>
      </button>
      <button
        aria-controls="panel-4"
        aria-selected={false}
        className="sc-dlfnuX egPkXS"
        id="tab-4"
        onClick={[Function]}
        role="tab"
        tabIndex={-1}
      >
        <span>
          Global Upwork
        </span>
      </button>
      <div
        className="sc-hKgJUU ktlWjT"
      />
    </div>
    <div
      className="sc-eCstlR jUdGBb"
    >
      <div
        aria-hidden={false}
        aria-labelledby="tab-0"
        className="sc-jSgvzq gYpxG"
        hidden={false}
        id="panel-0"
        role="tabpanel"
        tabIndex={0}
      >
        <h3>
          <span>
            Team Lead
          </span>
          <span
            className="company"
          >
             @ 
            <a
              className="inline-link"
              href="https://www.griddynamics.com/"
            >
              Grid Dynamics
            </a>
          </span>
        </h3>
        <p
          className="range"
        >
          MAR 2020 - Present
        </p>
        <div>
          <p>
            Building email sources & delivering optimization
          </p>
        </div>
        <div
          style={
            Object {
              "marginTop": 24,
            }
          }
        >
          <ul>
            <li>
              Team leading
            </li>
            <li>
              Improvement development flow
            </li>
            <li>
              Performance optimization
            </li>
          </ul>
        </div>
      </div>
      <div
        aria-hidden={true}
        aria-labelledby="tab-1"
        className="sc-jSgvzq gYpxG"
        hidden={true}
        id="panel-1"
        role="tabpanel"
        tabIndex={-1}
      >
        <h3>
          <span>
            Engineer
          </span>
          <span
            className="company"
          >
             @ 
            <a
              className="inline-link"
              href="https://qwil.io/"
            >
              Qwil Messenger
            </a>
          </span>
        </h3>
        <p
          className="range"
        >
          JUN 2020 - JAN 2021
        </p>
        <div>
          <p>
            Developing complex invitation tool with onboarding process.
          </p>
        </div>
        <div
          style={
            Object {
              "marginTop": 24,
            }
          }
        >
          <ul>
            <li>
              Code review
            </li>
            <li>
              Improving testing coverage
            </li>
            <li>
              Organization of meetups
            </li>
          </ul>
        </div>
      </div>
      <div
        aria-hidden={true}
        aria-labelledby="tab-2"
        className="sc-jSgvzq gYpxG"
        hidden={true}
        id="panel-2"
        role="tabpanel"
        tabIndex={-1}
      >
        <h3>
          <span>
            Fullstack
          </span>
          <span
            className="company"
          >
             @ 
            <a
              className="inline-link"
              href="https://codefactorygroup.com/"
            >
              Code Factory
            </a>
          </span>
        </h3>
        <p
          className="range"
        >
          JUN 2020 - JAN 2021
        </p>
        <div>
          <p>
            Developing a SportsBetting platform for the US market and back-office dashboard for internal use. Implemented RBAC/permissions features from scratch.
          </p>
        </div>
        <div
          style={
            Object {
              "marginTop": 24,
            }
          }
        >
          <ul>
            <li>
              Team lead
            </li>
            <li>
              Code review
            </li>
            <li>
              Monorepo management
            </li>
          </ul>
        </div>
      </div>
      <div
        aria-hidden={true}
        aria-labelledby="tab-3"
        className="sc-jSgvzq gYpxG"
        hidden={true}
        id="panel-3"
        role="tabpanel"
        tabIndex={-1}
      >
        <h3>
          <span>
            Team lead
          </span>
          <span
            className="company"
          >
             @ 
            <a
              className="inline-link"
              href="https://www.endava.com/"
            >
              Endava
            </a>
          </span>
        </h3>
        <p
          className="range"
        >
          JUN 2018 - June 2020
        </p>
        <div>
          <p>
            Development of a rich money exchange platform (Travelex) combining complex UI & business logic. The application makes use of KYC, sending money abroad, withdrawal, internal back-office management.
          </p>
        </div>
        <div
          style={
            Object {
              "marginTop": 24,
            }
          }
        >
          <ul>
            <li>
              Team leading
            </li>
            <li>
              Coaching
            </li>
            <li>
              Code review
            </li>
            <li>
              Migration from angular.js
            </li>
          </ul>
        </div>
      </div>
      <div
        aria-hidden={true}
        aria-labelledby="tab-4"
        className="sc-jSgvzq gYpxG"
        hidden={true}
        id="panel-4"
        role="tabpanel"
        tabIndex={-1}
      >
        <h3>
          <span>
            Fullstack
          </span>
          <span
            className="company"
          >
             @ 
            <a
              className="inline-link"
              href="https://www.upwork.com/"
            >
              Global Upwork
            </a>
          </span>
        </h3>
        <p
          className="range"
        >
          MAR 2013 - Present
        </p>
        <div>
          <p>
            Working on various custom projects. Building an impressive reputation on Upwork. Proven history of success with multiple clients.
          </p>
        </div>
        <div
          style={
            Object {
              "marginTop": 24,
            }
          }
        >
          <ul>
            <li>
              Top rated
            </li>
            <li>
              Rising talent
            </li>
          </ul>
        </div>
      </div>
    </div>
  </div>
</section>
`;
